src/Npgsql/NpgsqlConnector.cs

```diff
@@ -782,7 +782,7 @@ async Task ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationTok
                     if (perIpTimeout.IsSet)
                     {
                         combinedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-                        combinedCts.CancelAfter(perIpTimeout.TimeLeft.Milliseconds);
+                        combinedCts.CancelAfter((int)perIpTimeout.TimeLeft.TotalMilliseconds);
```

Comment:
That's quite a bug :rofl: 

We should check if we have this in 4.1 too...

Reply:
Just checked - 4.1 is safe.